packages/allow-scripts/src/setup.js

```diff
@@ -68,23 +67,54 @@ function writeRcFile (pkgManagerDefault = 'npm') {
   configs.forEach(writeRcFileContent)
 }
 
-function addPreinstallAFDependency () {
-  let cmd, cmdArgs
-
-  if (existsSync('./.npmrc')) {
-    cmd = 'npm'
-    cmdArgs = ['install', '-d', '@lavamoat/preinstall-always-fail']
-  } else {
-    cmd = 'yarn'
-    cmdArgs = ['add', '-D', '@lavamoat/preinstall-always-fail']
+function updatePackageJson(input) {
+  const p = path.join(process.cwd(), 'package.json')
+  const oldConf = JSON.parse(fs.readFileSync(p, { encoding: 'utf-8' }))
+  const mergedConf = { ...oldConf }
+  for (const [k, v] of Object.entries(input)) {
+    oldConf[k] = typeof v === 'object' && typeof oldConf[k] === 'object'
+      ? { ...oldConf[k], ...v }
+      : v
   }
+  fs.writeFileSyn(p, JSON.stringify(mergedConf, undefined, 2), { encoding: 'utf-8' })
+}
 
-  let result = spawnSync(cmd, cmdArgs, {})
+function patchPackageJson (runPkgManager) {
```

Comment:
Naming overload with `updatePackageJson`/`patchPackageJson`. TBD if they will be merged or either renamed.